pkg/deployment/deployment.go

```diff
@@ -141,9 +141,27 @@ func ConditionalDeploy(
 ) error {
 
 	var err error
+	var deployedConfigHash string
 	log := helper.GetLogger()
 
```

Comment:
ConditionalDeploy is called by Deploy. Deploy is only called from the Deployment reconciler. So what triggers this to start executing again?

I can't follow how updating the NodeSet, which updates the new Status fields on the NodeSet leads to the Deployment reconciling again.

Reply:
Ah yeah, I think the issue is that my env never set `instance.Status.Deployed = true` because it eventually got stuck on:
```
{"level":"error","ts":"2023-10-31T12:48:55+10:00","msg":"Reconciler error","controller":"openstackdataplanedeployment","controllerGroup":"dataplane.openstack.org","controllerKind":"OpenStackDataPlaneDeployment","OpenStackDataPlaneDeployment":{"name":"edpm-compute","namespace":"openstack-operators"},"namespace":"openstack-operators","name":"edpm-compute","reconcileID":"200e708b-b841-4eb1-9e0b-503bbbfce125","error":"ConfigMap \"ovncontroller-config\" not found","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/Users/brendanshephard/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:329\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/Users/brendanshephard/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:274\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/Users/brendanshephard/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:235"}
```

So you're right, if it had finished then my updates to the `NodeSet` would never be registered because we would exit the reconcile here:
https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/controllers/openstackdataplanedeployment_controller.go#L131-L134

So the only way this would work is if I were setting `instance.Status.Deployed` to `false` when the nodeSet changes. Which it currently doesn't do and requires more thought.